contracts/v3/FiatTokenV3.sol

```diff
@@ -0,0 +1,300 @@
+/**
+ * SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+pragma solidity 0.6.12;
+
+import { FiatTokenV2_1 } from "../v2/FiatTokenV2_1.sol";
+import { EIP712 } from "../util/EIP712.sol";
+
+// solhint-disable func-name-mixedcase
+
+/**
+ * @title FiatTokenV3
+ * @dev ERC20 Token backed by fiat reserves
+ */
+contract FiatTokenV3 is FiatTokenV2_1 {
+
+    /**
+     * @notice Initialize v3
+     */
+    function initializeV3(address[] calldata accountsToBlacklist) external {
```

Comment:
That's a good callout, I'll think about what the constraints are there (I think it would just be a tx gas constraint, e.g. how many blacklist calls you do within the gas limit of a transaction).

There are currently 41 addresses on the ETH L1 blacklist, which I think should be well within the constraint, but definitely worth knowing what the upper bound is.

Reply:
Thought about this some more--the gas footprint of each blacklist is mainly from the storage write, which costs 20K gas at worst (if the blacklisted user doesn't have a balance, only 2900 if there is a balance already there) for each blacklist call.

The size of Ethereum transactions is limited by block size.

Blocks can be bigger but 15M gas is the target block size. At 15M one could blacklist ~750 addresses in a single transaction assuming the worst case gas cost per address (but also assuming no other operations in the transaction). Taking into account the other checks performed in the upgrade transaction, let's call this ~700 address cap.

This should be more than adequate headspace given that USDC currently has 41 addrs on the blacklist.

![image](https://user-images.githubusercontent.com/13876978/182065962-1b9656e4-8d01-45d8-9fa8-6f0681639f43.png)

Reply:
well, now up to 81 addresses, but yeah still plenty of headroom.